src/main/kotlin/com/openlattice/hazelcast/serializers/SearchResultStreamSerializer.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+package com.openlattice.hazelcast.serializers
+
+import com.dataloom.mappers.ObjectMappers
+import com.hazelcast.nio.ObjectDataInput
+import com.hazelcast.nio.ObjectDataOutput
+import com.kryptnostic.rhizome.pods.hazelcast.SelfRegisteringStreamSerializer
+import com.openlattice.search.requests.SearchResult
+import com.fasterxml.jackson.core.type.TypeReference
+import com.openlattice.hazelcast.StreamSerializerTypeIds
+
+class SearchResultStreamSerializer(
+        private val hitType: TypeReference<List<Map<String, Any>>> = object : TypeReference<List<Map<String, Any>>>() {}
+) : SelfRegisteringStreamSerializer<SearchResult> {
+
+
+    override fun getTypeId(): Int {
+        return StreamSerializerTypeIds.SEARCH_RESULT.ordinal
+    }
+
+    override fun getClazz(): Class<out SearchResult> {
+        return SearchResult::class.java
+    }
+
+    override fun write(output: ObjectDataOutput, obj: SearchResult) {
+        output.writeLong(obj.numHits)
+        output.writeByteArray(ObjectMappers.getSmileMapper().writeValueAsBytes(obj.hits))
```

Comment:
It is my understanding that you don't need to use typereference in Kotlin as the type system will figure it out automatically for you.